src/dali/plugin/input/csv/bitbank_supplemental.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2022 macanudo527
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Withdrawal CSV Format: 日時 (timestamp), 数量 (amount), 手数料 (transaction fee), 合計 (total), ラベル (user-provided label), アドレス (address), Txid, ステータス (status)
+
+import logging
+from csv import reader
+from datetime import datetime
+from datetime import timezone as DatetimeTimezone
+from typing import List, Optional
+
+from pytz import timezone as PytzTimezone
+from rp2.logger import create_logger
+
+from dali.abstract_input_plugin import AbstractInputPlugin
+from dali.abstract_transaction import AbstractTransaction
+from dali.configuration import Keyword
+from dali.intra_transaction import IntraTransaction
+
+
+class InputPlugin(AbstractInputPlugin):
+
+    __BITBANK: str = "Bitbank"
+    __BITBANK_PLUGIN: str = "Bitbank_Supplemental_CSV"
+
+    __TIMESTAMP_INDEX: int = 0
+    __SENT_AMOUNT: int = 1
+    __TRANSACTION_FEE: int = 2
+    __TOTAL: int = 3
+    __LABEL: int = 4
+    __ADDRESS: int = 5
+    __TX_ID: int = 6
+    __STATUS: int = 7
+
+    __DELIMITER: str = ","
+
+    def __init__(
+        self,
+        account_holder: str,
+        withdrawals_csv_file: str,
+        withdrawals_code: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__withdrawals_csv_file: str = withdrawals_csv_file
+        self.__withdrawals_code: str = withdrawals_code
+        self.__logger: logging.Logger = create_logger(f"{self.__BITBANK_PLUGIN}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        return self.parse_withdrawals_file(self.__withdrawals_csv_file)
+
+    def parse_withdrawals_file(self, file_path: str) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(file_path, encoding="utf-8") as csv_file:
+            lines = reader(csv_file)
+
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                jst_timezone = PytzTimezone("Asia/Tokyo")
```

Comment:
Is this safe: i.e. Is this a Japan-only exchange?

Reply:
Yes, it"s Japan only. Japan has similar (but less restrictive) exchange rules. Any exchange that deals in JPY fiat must be registered in Japan and only in Japan. Of course, users are still free to transfer crypto off to another exchange that doesn't have such restrictions. These transactions have to be labeled though according to the "Travel Rule", so the Japanese government has a general idea of the flow of the assets.

This means a lot of the major exchanges have a special Japanese version (e.g. Coinbase has Coinbase Japan that operates independently from Coinbase US)

Reply:
I should also add that there is no timezone stamp, so we have to assume some time zone, unfortunately.

Reply:
Thanks for the explanation. The Trezor plugin had a similar problem (no timezone), so it lets the user specify it in the constructor: https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/csv/trezor.py#L52. 

There are two options: 
- the exchange assumes the timezone to be `Asia/Tokyo`: in this case there is nothing else to do
- the exchange records time with the user's local timezone as the implied timezone: in this case the Trezor approach can be useful (maybe have `Asia/Tokyo` as default value).

Do we know which of the two options applies here?

Reply:
That is a good question, but I'm going to guess that it will always be `Asia/Tokyo` at least while this exchange operates exclusively in Japan. Japan only has one timezone, so even if it was based on user's local timezone, it would probably not matter that much. Only Japanese residents can register for the exchange, so in order for the user's local timezone and server timezones to be different, it would have to be a Japanese resident travel overseas at the moment of the transaction. That does occur, but probably pretty rarely. 

Anyway, I can't seem to find any documentation on it. Their site doesn't contain any information as to whether it is definitively server time or local user's time. The REST API uses unix timestamps so doesn't have this issue at all.